File: exercise/11/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxarea = 0, left = 0, right = height.size() - 1;
        while (left < right) {
            int area = (right - left) * min(height[left], height[right]);
            if (area > maxarea) {
                maxarea = area;
            }
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxarea;
    }
};

File: exercise/11/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        result = 0
        while left < right:
            length = right - left
            edge = min(height[left], height[right])
            area = length * edge
            if result < area:
                result = area
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return result
